docs/spikes/2020-04-how-to-migrate-lb3-components.md

```diff
@@ -0,0 +1,626 @@
+# How to migrate LB3 components
+
+_GitHub issue: https://github.com/strongloop/loopback-next/issues/4099_
+
+## Out of scope
+
+Migration of the following component is out of scope of this spike:
+
+- [OAuth2](https://loopback.io/doc/en/lb3/OAuth-2.0.html) - will be covered by
+  auth&auth migration guide
+- [Passport](https://loopback.io/doc/en/lb3/Third-party-login-using-Passport.html) -
+  will be covered by auth&auth migration guide
+- [Synchronization](https://loopback.io/doc/en/lb3/Synchronization.html) - won't
+  be implemented in LB4. It's important to mention this fact in the migration
+  guide.
+
+## Table of contents
+
+- [Techniques](#techniques)
+  - [General](#general)
+  - [Models, Entities and Repositories](#models-entities-and-repositories)
+  - [REST API](#rest-api)
+  - [Services (local and remote)](#services-local-and-remote)
+  - [API transports](#api-transports)
+  - [Authentication & authorization](#authentication--authorization)
+  - [Introspection](#introspection)
+- [Migration of component layout & instructions from mounting to an app](#migration-of-component-layout--instructions-from-mounting-to-an-app)
+  - [LB3 component layout & mounting](#lb3-component-layout--mounting)
+  - [Migration to LB4](#migration-to-lb4)
+- [Overview of existing LB3 components](#overview-of-existing-lb3-components)
+  - [Push notifications](#push-notifications)
+  - [Storage component](#storage-component)
+  - [Logging components](#logging-components)
+  - [Access control](#access-control)
+  - [Connectors](#connectors)
+  - [Custom transports](#custom-transports)
+  - [Inspecting models](#inspecting-models)
+  - [Unsorted](#unsorted)
+  - [Mixins](#mixins)
+
+## Techniques
+
+### General
+
+1. How to structure the extension code to receive user-provided configuration &
+   target app instance. What are the instructions for adding a LB4 component to
+   a LB4 app.
```

Comment:
It's not obvious as there is no explanation of LB3 component's signature `(app, config, callback)`.